### DISCUSSION TITLE
Motion planning on mujoco

### DISCUSSION THREAD

**Discussion by gabrielpeixoto-cvai** [google-deepmind/mujoco#2094] [Asking for Help]
### The feature, motivation and pitch

Hello, First of all, thanks for the great work and sharing this with everyone!

I am doing research in motion planning with collision avoidance. I have been using Bullet and OMPL to simulate robots. I want to switch to mujoco, so I wondered how to achieve a similar setup.

I found this [repository](https://github.com/mpflueger/mujoco-ompl), but it seems to be old (before mujoco was open-sourced) so probably there should be a lot of work in adapting it to the current mujoco version.

I am searching for information. What I need is the following:

* IK: which I know it is available in [dm_control](https://github.com/google-deepmind/dm_control)
* Collision check: I think mujoco already has the contacts available;
* Motion planning algorithms: for example, RRT, RRT*, and others;

Making an OMPL wrapper that works with mujoco would be the best.

### Alternatives

I could not find many resources on this topic.
I know that [mujoco_mpc](https://github.com/google-deepmind/mujoco_mpc) exists, but it does not satisfy my requirements.

**Comment by yuvaltassa** [comment]
Regarding IK, you should probably look into @kevinzakka's [mink](https://github.com/kevinzakka/mink), the dm_control stuff is not great.

Regarding an OMPL wrapper for MuJoCo, that would great! If you decide to write it, we will be happy to help 🙂

**Reply by gabrielpeixoto-cvai** [reply]
@yuvaltassa Thanks for the quick response! 
I will check @kevinzakka repo for IK and I will evaluate the [mujoco-ompl](https://github.com/mpflueger/mujoco-ompl) repository I mentioned on the post, 
I will update this discussion when I have updates on the progress.

**Reply by jf514** [reply]
@gabrielpeixoto-cvai Just curious, why is Mujoco a better solution for you than others, such as MoveIt?

**Reply by gabrielpeixoto-cvai** [reply]
@jf514 In short, speed, Python support, and soft body simulation.

I have ROS1 and pybullet environments with the capabilities mentioned in my post. ROS needs too much boilerplate and cannot simulate soft bodies (I am not sure about the gazebo; I never used it because it was too cumbersome to set up). Pybullet softbody simulation is somewhat broken.

I will eventually setup a ROS2 + Moveit system for controlling real robots, but I still want to prototype with Mujoco at first.

**Comment by ProNeverFake** [comment]
I'm also looking for mujoco motion planning solutions for my simulation data pipeline. If you want to write an OMPL wrapper I would be glad to help ;)

**Comment by shaoyifei96** [comment]
Also happy to help if this is gonna be a thing

**Comment by Tadinu** [comment]
@gabrielpeixoto-cvai Hope these are useful to you
https://github.com/Tadinu/awesome_mujoco/tree/main?tab=readme-ov-file#motion-planning
https://github.com/joonhyung-lee/mujoco-robotics-usage
https://github.com/sjchoi86/yet-another-mujoco-tutorial-v3
